src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
Ah I see, yes I believe this was @Mark-Simulacrum's original suggested approach. On reflection, this is probably doable with no more boilerplate than wrapping the struct definition in a macro.

(also, for some reason I previously thought you couldn't have per-function static variables which didn't help my thinking)